lib/inspec/resources/mysql_session.rb

```diff
@@ -68,6 +72,12 @@ def to_s
 
     private
 
+    # Querying on the database to make sure conneciton can be established. If not this will set the resource exception
+    # message which we raise before querying on the database using mysql_session object.
+    def set_connection
```

Comment:
I suggest we rename this to be "test connection" and use a query like "SELECT 1" - some users might not have permission to list databases.